site/src/modules/notifications/NotificationsInbox/NotificationsInbox.stories.tsx

```diff
@@ -0,0 +1,178 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { expect, fn, userEvent, within } from "@storybook/test";
+import { MockNotifications, mockApiError } from "testHelpers/entities";
+import { withGlobalSnackbar } from "testHelpers/storybook";
+import { NotificationsInbox } from "./NotificationsInbox";
+
+const meta: Meta<typeof NotificationsInbox> = {
+	title: "modules/notifications/NotificationsInbox/NotificationsInbox",
+	component: NotificationsInbox,
+	render: (args) => {
+		return (
+			<div className="w-full max-w-screen-xl p-6">
+				<header className="flex justify-end">
+					<NotificationsInbox {...args} />
+				</header>
+			</div>
+		);
+	},
+};
+
+export default meta;
+type Story = StoryObj<typeof NotificationsInbox>;
+
+export const Default: Story = {
+	args: {
+		defaultOpen: true,
+		fetchNotifications: fn(() =>
+			Promise.resolve({ notifications: MockNotifications, unread_count: 2 }),
+		),
+	},
+};
+
+export const Failure: Story = {
+	args: {
+		defaultOpen: true,
+		fetchNotifications: fn(() =>
+			Promise.reject(
+				mockApiError({
+					message: "Failed to load notifications",
+				}),
+			),
+		),
+	},
+};
+
+export const FailAndRetry: Story = {
+	args: {
+		defaultOpen: true,
+		fetchNotifications: (() => {
+			let count = 0;
+
+			return fn(() => {
+				count += 1;
+
+				// Fail on the first 3 attempts
+				// 3 is the maximum number of retries from react-query
```

Comment:
so just to make sure I understand, the third retry is the fourth request, right? which will then fall through to the resolve

Reply:
I think instead of retries I should have use "tries". react-query tries 3x times to get a success response.